src/org/wso2/tg/jenkins/executors/TestExecutor.groovy

```diff
@@ -162,6 +172,39 @@ def prepareWorkspace(testPlanId, scenarioConfigs) {
     }
 }
 
+/**
+ * Returns the parallel id of the test test plan
+ *
+ * @param name of the executor where test plan is running
+ * @return parallel id of test plan
+ */
+@NonCPS
+def getParalleId(name){
+
+    def parallelname = name
+    def build = currentBuild.getRawBuild()
+    def actions = build.getAllActions();
+    def parallelId = ""
+    actions.each{act ->
+        if(act instanceof FlowGraphAction){
+            def nodes = act.getNodes();
+            nodes.each{node ->
+                if(node instanceof StepStartNode){
+                    if(node.displayName.endsWith(parallelname)){
+                        println("Parallel node  : " + parallelname + " : " + node.id.toString())
+                        parallelId = node.id.toString()
+                    }
+                }
+
+            }
+        }
+    }
+    build = null
```

Comment:
Shall we add a  comment why you are doing this?

Reply:
+1